lib/matplotlib/colors.py

```diff
@@ -1127,61 +1127,67 @@ class DivergingNorm(TwoSlopeNorm):
     ...
 
 
+def _make_norm_from_scale(scale_cls, base_cls=None, *, init=None):
+    if base_cls is None:
+        return functools.partial(_make_norm_from_scale, scale_cls, init=init)
+
+    if init is None:
+        def init(vmin=None, vmax=None, clip=False): pass
+    init_signature = inspect.signature(init)
+
+    class Norm(base_cls):
+
+        def __init__(self, *args, **kwargs):
+            ba = init_signature.bind(*args, **kwargs)
+            ba.apply_defaults()
+            super().__init__(
+                **{k: ba.arguments.pop(k) for k in ["vmin", "vmax", "clip"]})
+            self._scale = scale_cls(axis=None, **ba.arguments)
```

Comment:
This looks great to me.  For colorbar we will need access to `self._scale`, which I assume this is where we get this.

Reply:
Yes.